CRM/Utils/System/Drupal8.php

```diff
@@ -835,4 +835,19 @@ public function getCMSPermissionsUrlParams() {
     return ['ufAccessURL' => \Drupal\Core\Url::fromRoute('user.admin_permissions')->toString()];
   }
 
+  /**
+   * Start a new session.
+   */
+  public function sessionStart() {
+    if (\Drupal::hasContainer()) {
+      $session = \Drupal::request()->getSession();
```

Comment:
It's better to grab from `session` service than the request session. I've seen some weirdness otherwise. Can't fully explain why, as kids usually the same reference

Reply:
Okay, the parent class has this:

```

  /**
   * Start a new session.
   */
  public function sessionStart() {
    session_start();
  }
```

If you look at the session manager in Drupal, which is what `$session->start()` calls, you'll see it's safe for existing started sessions: https://git.drupalcode.org/project/drupal/-/blob/8.8.x/core/lib/Drupal/Core/Session/SessionManager.php#L106

This is regenerate, which migrate calls: https://git.drupalcode.org/project/drupal/-/blob/8.8.x/core/lib/Drupal/Core/Session/SessionManager.php#L207

I don't know if this code actually should call migrate over `->start()` directly.